### PR TITLE
Save region correctly to save sales address from admin [backport]

### PR DESCRIPTION
**Pull request by raumatbel**
Related with [PR#11381](https://github.com/magento/magento2/pull/11381)
### Description

Save region correctly to save sales address from admin
### Fixed Issues (if relevant)

1. magento/magento2#10441: State/Province Not displayed after edit Billing Address on Sales Orders - Backend Admin.

### Steps to reproduce
1. Go to the backend -> Sales -> Orders -> View any order.
2. Select the section -> Information -> Address Information
3. Edit Billing Address, for example the street address.
4. Save Order Address.

**Note:** The same issue happen with the Shipping Address.
### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by okorshenko** [comment]
Hi @raumatbel I just sent you an invitation to magento/magento2 repo. Could you please accept it

**Comment by raumatbel** [comment]
@okorshenko Accepted

**Comment by raumatbel** [comment]
@okorshenko I have a problem with travis. The error shown is the next.
travis
In my code, in the line 18, I only extend the class (similar like the origin file)
class
What is the best solution to solve this?

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php [hunk 1]
```diff
@@ -4,9 +4,32 @@
...
+use Magento\Framework\Translate\InlineInterface;
+use Magento\Framework\View\Result\PageFactory;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\View\Result\LayoutFactory;
+use Magento\Framework\Controller\Result\RawFactory;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ * @SuppressWarnings(PHPMD.ExcessiveParameterList)
```
Comment (vkublytskyi) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php]: `@SuppressWarnings(PHPMD.ExcessiveParameterList)` should be used for the constructor.

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php [hunk 2]
```diff
@@ -16,18 +39,69 @@ class AddressSave extends \Magento\Sales\Controller\Adminhtml\Order
...
+        FileFactory $fileFactory,
+        InlineInterface $translateInline,
+        PageFactory $resultPageFactory,
+        JsonFactory $resultJsonFactory,
+        LayoutFactory $resultLayoutFactory,
+        RawFactory $resultRawFactory,
+        OrderManagementInterface $orderManagement,
+        OrderRepositoryInterface $orderRepository,
+        LoggerInterface $logger,
+        RegionFactory $regionFactory
```
Comment (vkublytskyi) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php]: Please use approach for extending constructor arguments in [backward compatible way](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter)

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php [hunk 3]
```diff
@@ -51,4 +125,19 @@ public function execute()
...
         }
     }
+    
+    /**
+     * Update region data
+     *
+     * @param array $attributeValues
+     * @return void
+     */
+    protected function updateRegionData(&$attributeValues)
```
Comment (vkublytskyi) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php]: Passing arguments by reference is not recommended. Return modified data instead (possible memory usage optimization is not so important here).

**Review by vkublytskyi — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php [hunk 4]
```diff
@@ -51,4 +126,20 @@ public function execute()
...
         }
     }
+    
+    /**
+     * Update region data
+     *
+     * @param array $attributeValues
+     * @return array
+     */
+    protected function updateRegionData($attributeValues)
```
Comment (vrann) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php]: Why this method protected? Is it referenced from the children classes?

Comment (raumatbel) [reply app/code/Magento/Sales/Controller/Adminhtml/Order/AddressSave.php]: @vrann this function is only used in this class. I have updated the visibility to private.

**Pull request merged**